### ISSUE TITLE
D and G loss stop changing. Hyperparameters anyone?

### ISSUE THREAD

**Issue by Ianmcmill** [DataSnaek/DCGAN-Keras#2]
Windows 10
All dependencies uptodate.
Arguments:
`python DCGAN.py --data ./data/portrait_large_homogen_256/*.jpg --batch_size 8 --image_size (256,256) --epochs 500000 --save_interval 500 --output_directory ./data/output`

Dataset consists of ~6300 jpeg art portrait taken from the wikiart dataset. All resized to 256x256 RGB.
After some epochs the losses stall.

```
13882 [D loss: 7.971192359924316 | D Accuracy: 50.0] [G loss: 1.192093321833454e-07]
13883 [D loss: 7.971192359924316 | D Accuracy: 50.0] [G loss: 1.192093321833454e-07]
13884 [D loss: 7.971192359924316 | D Accuracy: 50.0] [G loss: 1.192093321833454e-07]
13885 [D loss: 7.971192359924316 | D Accuracy: 50.0] [G loss: 1.192093321833454e-07]
13886 [D loss: 7.971192359924316 | D Accuracy: 50.0] [G loss: 1.192093321833454e-07]
13887 [D loss: 7.971192359924316 | D Accuracy: 50.0] [G loss: 1.192093321833454e-07]
13888 [D loss: 7.971192359924316 | D Accuracy: 50.0] [G loss: 1.192093321833454e-07]
13889 [D loss: 7.971192359924316 | D Accuracy: 50.0] [G loss: 1.192093321833454e-07]
13890 [D loss: 7.971192359924316 | D Accuracy: 50.0] [G loss: 1.192093321833454e-07]
13891 [D loss: 7.971192359924316 | D Accuracy: 50.0] [G loss: 1.192093321833454e-07]
13892 [D loss: 7.971192359924316 | D Accuracy: 50.0] [G loss: 1.192093321833454e-07]
13893 [D loss: 7.971192359924316 | D Accuracy: 50.0] [G loss: 1.192093321833454e-07]
13894 [D loss: 7.971192359924316 | D Accuracy: 50.0] [G loss: 1.192093321833454e-07]
13895 [D loss: 7.971192359924316 | D Accuracy: 50.0] [G loss: 1.192093321833454e-07]
13896 [D loss: 7.971192359924316 | D Accuracy: 50.0] [G loss: 1.192093321833454e-07]
13897 [D loss: 7.971192359924316 | D Accuracy: 50.0] [G loss: 1.192093321833454e-07]
13898 [D loss: 7.971192359924316 | D Accuracy: 50.0] [G loss: 1.192093321833454e-07]
13899 [D loss: 7.971192359924316 | D Accuracy: 50.0] [G loss: 1.192093321833454e-07]
13900 [D loss: 9.985954284667969 | D Accuracy: 37.5] [G loss: 1.192093321833454e-07]
13901 [D loss: 7.971192359924316 | D Accuracy: 50.0] [G loss: 1.192093321833454e-07]
13902 [D loss: 7.971192359924316 | D Accuracy: 50.0] [G loss: 1.192093321833454e-07]
```

Second question: I really wonder why I can run this with 256x output_size and batch_size 8 (higher not tested) on my gtx 980 4gb. And at that speed...
All other dcgan codes here on github fail with such a network.

**Comment by Ianmcmill** [comment]
The image sizes are too big for this GAN. The two networks diverge pretty quickly. Lowering the resolution gives better results.
256x256 portrait dataset
5000 epochs
![5000](https://user-images.githubusercontent.com/10505658/47935322-4a902e80-deda-11e8-9be5-71280afc3250.jpg)

96x96 portrait dataset
5000 epochs
![5000](https://user-images.githubusercontent.com/10505658/47935350-5f6cc200-deda-11e8-8851-c192dd5d9e85.jpg)

12000 epochs
![12000](https://user-images.githubusercontent.com/10505658/47935384-6f84a180-deda-11e8-89de-94b3e957a656.jpg)

**Comment by DataSnaek** [comment]
When training on higher resolutions such as 256x, don't expect good results in a small amount of time. It took about 80,000 epochs to get good results for me at 256x. Think about it, there's far more parameters to optimise in a network that large.

Here's what my 256x landscapes looked like at 5,000 epochs:

![4950](https://user-images.githubusercontent.com/36249164/48085645-bf24df00-e1f2-11e8-89a5-ff86ef05eeee.png)

Compared to some after 80,000:

![0](https://user-images.githubusercontent.com/36249164/48085736-06ab6b00-e1f3-11e8-9a9a-05e550791fc6.png)

As for why you can run this network at high resolution with that batch size, it's most likely because of the self.starting_filters value I have chosen. Different networks do things different ways, but because that number is by default relatively low in my code, and because I made many upsample layers by default, the memory consumption is lower. 

Doing so comes at the cost of image quality I assume, but it's a balance of wanting higher resolution images, time and quality.

**Comment by DataSnaek** [comment]
As for the converging networks, the parameters to try and change that will have the most impact are the self.starting_filters value, the various filter sizes of Generator and Discriminator layers and the number of upsample layers in each of them (and then set the self.upsample_layers value to match)

All of these have big effects on memory usage as well though, so be careful.
